### PR TITLE
Misc small nix improvements

### PR DESCRIPTION
**Pull request by rvl**
- Three small nix tweaks.
- Fix evaluation error which was preventing bors from merging.
- Remove the Buildkite hydra-eval-errors job in favour of github ci/hydra-eval status. It should be much quicker and easier to spot nix errors with this.

[Hydra Jobset](https://hydra.iohk.io/jobset/Cardano/cardano-wallet-pr-1661)

### CODE REVIEW

File: nix/util.nix [hunk 1]
```diff
@@ -21,4 +21,15 @@ with lib; with haskell-nix.haskellLib;
...
+
+  # Convert version strings from Cabal format (YYYY.M.D)
+  # to git tag format (vYYYY-MM-DD).
+  versionTag = cabal: let
+      parts = builtins.match "([[:digit:]]{4})\.([[:digit:]]{1,2})\.([[:digit:]]{1,2})" cabal;
+      leading0 = str: if stringLength str == 1 then "0" + str else str;
+    in
+      assert assertMsg (parts != null)
+        "versionTag: ${cabal} is not in YYYY.M.D format";
+      "v" + concatMapStringsSep "-" leading0 parts;
```
Comment (KtorZ) [inline nix/util.nix:33]: This is simply a function declaration, not used anywhere yet, right? We have to be careful no breaking the Travis setup.

Comment (rvl) [reply nix/util.nix:33]: Yes it's not used yet, but it could be used to make filenames more consistent, and remove the need to rename files in Travis.

**Comment by rvl** [comment]
bors try

**Comment by rvl** [comment]
bors try

**Pull request merged**